test/cases/regression-old-worker-new-composer.sh

```diff
@@ -54,123 +59,362 @@ priority=${priority}
 EOF
 }
 
-# Composer v33
+# Composer v51
+greenprint "Installing osbuild-composer-worker from commit ${WORKER_VERSION}"
 setup_repo osbuild-composer "$WORKER_VERSION" 20
-sudo dnf install -y osbuild-composer-worker podman composer-cli
+sudo dnf install -y osbuild-composer-worker osbuild-composer-dnf-json podman composer-cli
 
 # verify the right worker is installed just to be sure
 rpm -q "$WORKER_RPM"
 
-# run container
-WELDR_DIR="$(mktemp -d)"
-WELDR_SOCK="$WELDR_DIR/api.socket"
-DNF_DIR="$(mktemp -d)"
-DNF_SOCK="$DNF_DIR/api.sock"
+if which podman 2>/dev/null >&2; then
+  CONTAINER_RUNTIME=podman
+elif which docker 2>/dev/null >&2; then
+  CONTAINER_RUNTIME=docker
+else
+  echo No container runtime found, install podman or docker.
+  exit 2
+fi
+
+
+# Container image used for cloud provider CLI tools
+CONTAINER_IMAGE_CLOUD_TOOLS="quay.io/osbuild/cloud-tools:latest"
 
-sudo podman pull --creds "${V2_QUAY_USERNAME}":"${V2_QUAY_PASSWORD}" \
+greenprint "Pulling and running composer container for this commit"
+sudo ${CONTAINER_RUNTIME} pull --creds "${V2_QUAY_USERNAME}":"${V2_QUAY_PASSWORD}" \
      "quay.io/osbuild/osbuild-composer-ubi-pr:${CI_COMMIT_SHA}"
 
+cat <<EOF | sudo tee "/etc/osbuild-composer/osbuild-composer.toml"
+log_level = "debug"
+[koji]
+allowed_domains = [ "localhost", "client.osbuild.org" ]
+ca = "/etc/osbuild-composer/ca-crt.pem"
+[koji.aws_config]
+bucket = "${AWS_BUCKET}"
```

Comment:
This makes me realize that another reason to update the worker in the regression test here is configs. So the bucket in the hosted service is configured in the worker itself, I'll be opening a PR soon to drop the bucket config from the cloudapi. So it's fine here because the worker has supported this for a while now, but in future we might need to update the worker here to benefit from updated configs (multi-region support is one of the things coming up).

Reply:
Good to know.  Should definitely update these.  It's not rare that I'll look at tests (instead of docs :upside_down_face:) to figure out how to configure something, so the tests should do things properly.